main.go

```diff
@@ -0,0 +1,296 @@
+	application.Flag("failmissing", "Fail if a directory in the hierarchy is missing.").Short('m').
+		Envar("HIERARCHY_FAILMISSING").Default("false").BoolVar(&cfg.failMissing)
+	application.Flag("debug", "Print debug output.").Short('d').
+		Envar("HIERARCHY_DEBUG").Default("false").BoolVar(&cfg.logDebug)
+	application.Flag("trace", "Prints a diff after processing each file. This generates A LOT of output.").
+		Envar("HIERARCHY_TRACE").Default("false").BoolVar(&cfg.logTrace)
+	application.Flag("version", "Print version and build information, then exit.").Short('V').
+		Default("false").BoolVar(&cfg.printVersion)
+
+	_, err := application.Parse(os.Args[1:])
+
+	if cfg.printVersion {
+		version.Print()
+		os.Exit(0)
+	}
+
+	if err != nil {
+		fmt.Fprintln(os.Stderr, errors.Wrapf(err, "Error parsing command-line arguments"))
+		application.Usage(os.Args[1:])
+		os.Exit(2)
+	}
+	return cfg
+}
+
+// checkForError fails the program with a fatal error message if e != nil
+func checkForError(e error) {
+	if e != nil {
+		log.Fatal(e)
+	}
+}
+
+// processHierarchy loads the hierarchy file and generates a list of file paths
+// of folders to be processed
+func processHierarchy(cfg config) []string {
+	hierarchy := []string{}
+	hierarchyFile, err := os.Open(cfg.hierarchyFile)
+	checkForError(err)
+	defer hierarchyFile.Close()
+
+	// Start reading from the file with a reader
+	reader := bufio.NewReader(hierarchyFile)
+	var line string
+	for {
+		line, err = reader.ReadString('\n')
+		if err != nil && err != io.EOF {
+			break
+		}
+
+		// Trim spaces and comments
+		includePath := strings.Split(line, "#")[0]
+		includePath = strings.TrimSpace(includePath)
+		includePath = replaceEnvironmentVariables(includePath)
+		// Process path
+		if len(includePath) > 0 {
+			includePath = path.Join(cfg.basePath, includePath)
+			// Check if directory exists
+			if stat, err := os.Stat(includePath); err == nil && stat.IsDir() {
+				hierarchy = append(hierarchy, includePath)
+				absPath, _ := filepath.Abs(includePath)
+				log.WithFields(log.Fields{
+					"path":     includePath,
+					"abs_path": absPath,
+				}).Debug("Adding path to hierarchy")
+			} else {
+				if cfg.failMissing {
+					log.WithFields(log.Fields{
+						"path": includePath,
+					}).Fatal("Hierarchy directory not found")
+				} else {
+					log.WithFields(log.Fields{
+						"path": includePath,
+					}).Warning("Ignoring missing hierarchy directory")
+				}
+			}
+		}
+
+		// Break the for loop on error including EOF
+		if err != nil {
+			break
+		}
+	}
+	if err != io.EOF {
+		checkForError(err)
+	}
+	return hierarchy
+}
+
+// mergeYamls walks through all the folders in the hierarchy
+// and merges all files matching the pattern into the structure,
+// overwriting any existing values
+// and exports the merged content to a new YAML file
+func mergeYamls(hierarchy []string, fileFilter string, outputFile string) {
```

Comment:
```suggestion
func mergeFilesInHierarchy(hierarchy []string, fileFilter string, outputFile string) {
```

Reply:
No longer is just merging Yamls, so think func should be renamed. Would need to change naming in all refs if accept change for this one